### PR TITLE
feat(knock) : Knock Requests Banner UI

### PR DESCRIPTION
**Pull request by ganfra**
## Content
First iteration of the knock requests [banner view](https://www.figma.com/design/7TqjqdMBaPpm3IavKsMYu6/Ask-to-join-(Knocking)?node-id=3162-29638&m=dev)
No data branched. 

## Motivation and context
Part of https://github.com/element-hq/element-x-android/issues/3692

## Tests

- Step 1
- Step 2
- Step ...

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes have been tested on an Android device or Android emulator with API 23
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/element-hq/element-x-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request title will be used in the release note, it clearly define what will change for the user
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR

### CODE REVIEW

File: features/knockrequests/impl/src/main/kotlin/io/element/android/features/knockrequests/impl/banner/KnockRequestsBannerPresenter.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import androidx.compose.runtime.setValue
+import io.element.android.libraries.architecture.AsyncAction
+import io.element.android.libraries.architecture.Presenter
+import kotlinx.collections.immutable.persistentListOf
+import javax.inject.Inject
+
+class KnockRequestsBannerPresenter @Inject constructor() : Presenter<KnockRequestsBannerState> {
+    @Composable
+    override fun present(): KnockRequestsBannerState {
+        var shouldShowBanner by remember { mutableStateOf(false) }
```
Comment (bmarty) [inline features/knockrequests/impl/src/main/kotlin/io/element/android/features/knockrequests/impl/banner/KnockRequestsBannerPresenter.kt:23]: The banner will be shown at each app starts? Or will the app will store the decision to hide the banner in a later PR?

File: features/knockrequests/impl/src/main/kotlin/io/element/android/features/knockrequests/impl/banner/KnockRequestsBannerState.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    val canAccept: Boolean,
+    val eventSink: (KnockRequestsBannerEvents) -> Unit,
+) {
+    val subtitle = if (knockRequests.size == 1) {
+        knockRequests.first().userId.value
+    } else {
+        null
+    }
+
+    val reason = if (knockRequests.size == 1) {
```
Comment (bmarty) [inline features/knockrequests/impl/src/main/kotlin/io/element/android/features/knockrequests/impl/banner/KnockRequestsBannerState.kt:33]: Could be nice to have a `firstIfSingleton` extension.

File: features/knockrequests/impl/src/main/kotlin/io/element/android/features/knockrequests/impl/banner/KnockRequestsBannerView.kt [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+            } else {
+                OutlinedButton(
+                    text = "View",
+                    onClick = onViewRequestsClick,
+                    size = ButtonSize.MediumLowPadding,
+                    modifier = Modifier.weight(1f),
+                )
+                if (state.canAccept) {
+                    Button(
+                        text = "Accept",
```
Comment (bmarty) [inline features/knockrequests/impl/src/main/kotlin/io/element/android/features/knockrequests/impl/banner/KnockRequestsBannerView.kt:150]: Use `stringResource` please (several changes to do in this place)

File: features/knockrequests/impl/src/main/kotlin/io/element/android/features/knockrequests/impl/banner/KnockRequestsBannerView.kt [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+                        onClick = {},
```
Comment (bmarty) [inline features/knockrequests/impl/src/main/kotlin/io/element/android/features/knockrequests/impl/banner/KnockRequestsBannerView.kt:151]: Can be done later, but maybe prepare the lambda for the onClick action.

File: tests/uitests/src/test/snapshots/images/features.knockrequests.impl.banner_KnockRequestsBannerView_Night_5_en.png
Comment (bmarty) [inline tests/uitests/src/test/snapshots/images/features.knockrequests.impl.banner_KnockRequestsBannerView_Night_5_en.png:1]: Show a loading wheel and disable the button here? (Loading state)

**Review by bmarty — APPROVED** [review]
A few remarks, else LGTM

File: features/knockrequests/impl/src/main/kotlin/io/element/android/features/knockrequests/impl/banner/KnockRequestsBannerPresenter.kt (hunk 1 above)
Comment (ganfra) [reply features/knockrequests/impl/src/main/kotlin/io/element/android/features/knockrequests/impl/banner/KnockRequestsBannerPresenter.kt:23]: Will be stored by the sdk actually!

File: tests/uitests/src/test/snapshots/images/features.knockrequests.impl.banner_KnockRequestsBannerView_Night_5_en.png
Comment (ganfra) [reply tests/uitests/src/test/snapshots/images/features.knockrequests.impl.banner_KnockRequestsBannerView_Night_5_en.png:1]: The banner will be dismissed directly in this case, I'm waiting to branch the logic

**Pull request merged**